### PR TITLE
docs: translate Thinking in React

### LINKED ISSUE #374: Thinking in React
Page: https://id.react.dev/learn/thinking-in-react

Assignee: @resir014

### PR DESCRIPTION
**Pull request by resir014**
Translated the "Thinking in React" page to Indonesian.

Closes #374.

### CODE REVIEW

**Comment by resir014** [comment]
@mhaidarhanif @r17x @zainfathoni Ready to review.

File: src/content/learn/thinking-in-react.md [hunk 1]
```diff
@@ -1,18 +1,18 @@
 ---
-title: Thinking in React
+title: Cara Berpikir dengan React
 ---
 
 <Intro>
 
-React can change how you think about the designs you look at and the apps you build. When you build a user interface with React, you will first break it apart into pieces called *components*. Then, you will describe the different visual states for each of your components. Finally, you will connect your components together so that the data flows through them. In this tutorial, we’ll guide you through the thought process of building a searchable product data table with React.
+React dapat mengubah cara berpikir Anda tentang desain yang Anda lihat dan aplikasi yang Anda buat. Ketika Anda membuat antarmuka pengguna dengan React, pertama-tama Anda akan memecahnya menjadi beberapa bagian yang disebut dengan *komponen*. Kemudian, Anda akan mendeskripsikan state visual yang berbeda untuk setiap komponen Anda. Terakhir, Anda akan menghubungkan komponen-komponen Anda bersama-sama sehingga data mengalir melaluinya. Dalam tutorial ini, kami akan memandu Anda melalui proses berpikir untuk membangun tabel data produk yang dapat dicari dengan React.
```
Comment (mhaidarhanif) [inline src/content/learn/thinking-in-react.md:7]: `state` need to be italic? Also some people said was confused when being introduced with the name `antarmuka pengguna`.

```suggestion
React dapat mengubah cara berpikir Anda tentang desain yang Anda lihat dan aplikasi yang Anda buat. Ketika Anda membuat antarmuka pengguna (*user interface*) dengan React, pertama-tama Anda akan memecahnya menjadi beberapa bagian yang disebut dengan *komponen*. Kemudian, Anda akan mendeskripsikan *state* visual yang berbeda untuk setiap komponen Anda. Terakhir, Anda akan menghubungkan komponen-komponen Anda bersama-sama sehingga data mengalir melaluinya. Dalam tutorial ini, kami akan memandu Anda melalui proses berpikir untuk membangun tabel data produk yang dapat dicari dengan React.
```

File: src/content/learn/thinking-in-react.md [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
 
 </Intro>
 
-## Start with the mockup {/*start-with-the-mockup*/}
+## Mulailah dengan sebuah rancang bangun {/*start-with-the-mockup*/}
 
-Imagine that you already have a JSON API and a mockup from a designer.
+Bayangkan Anda sudah memiliki API JSON dan rancang bangun dari desainer.
```
Comment (mhaidarhanif) [inline src/content/learn/thinking-in-react.md:13]: To comply with the source, `from a designer` with `a`.

```suggestion
Bayangkan Anda sudah memiliki API JSON dan rancang bangun dari seorang desainer.
```

File: src/content/learn/thinking-in-react.md [hunk 3]
```diff
@@ -1,18 +1,18 @@
...
 
-The JSON API returns some data that looks like this:
+API JSON mengembalikan beberapa data yang terlihat seperti ini:
```
Comment (mhaidarhanif) [inline src/content/learn/thinking-in-react.md:15]: Do we need to add this? To make the reading flow better:

```suggestion
API JSON tersebut mengembalikan beberapa data yang terlihat seperti ini:
```

File: src/content/learn/thinking-in-react.md [hunk 4]
```diff
@@ -25,59 +25,59 @@ The JSON API returns some data that looks like this:
 ]
 ```
 
-The mockup looks like this:
+Rancang bangun kita terlihat seperti ini:
```
Comment (mhaidarhanif) [inline src/content/learn/thinking-in-react.md:28]: Similar to the previous sentence. Also, the source is not `Our mockup`, but `The mockup`.

```suggestion
Rancang bangun tersebut terlihat seperti ini:
```

**Review by mhaidarhanif — CHANGES_REQUESTED** [review]
Some early suggestions and feedback to discuss.

File: src/content/learn/thinking-in-react.md [hunk 5]
```diff
@@ -25,59 +25,59 @@ The JSON API returns some data that looks like this:
...
 <img src="/images/docs/s_thinking-in-react_ui.png" width="300" style={{margin: '0 auto'}} />
 
-To implement a UI in React, you will usually follow the same five steps.
+Untuk mengimplementasikan antarmuka pengguna di React, Anda biasanya akan mengikuti lima langkah yang sama.
 
-## Step 1: Break the UI into a component hierarchy {/*step-1-break-the-ui-into-a-component-hierarchy*/}
+## Langkah 1: Bagi antarmuka pengguna menjadi hierarki komponen {/*step-1-break-the-ui-into-a-component-hierarchy*/}
 
-Start by drawing boxes around every component and subcomponent in the mockup and naming them. If you work with a designer, they may have already named these components in their design tool. Ask them!
+Mulailah dengan menggambar kotak-kotak di sekitar setiap komponen dan subkomponen dalam rancang bangun dan beri nama. Jika Anda bekerja dengan seorang desainer, mereka mungkin telah menamai komponen-komponen ini di alat bantu desain mereka. Tanyakan kepada mereka!
```
Comment (mhaidarhanif) [inline src/content/learn/thinking-in-react.md:36]: To make the reading flow better.

```suggestion
Mulailah dengan menggambar kotak-kotak di sekitar setiap komponen dan subkomponen dalam rancang bangun dan berikan nama. Jika Anda bekerja dengan seorang desainer, mereka mungkin telah menamai komponen-komponen ini di alat bantu desain mereka. Tanyakanlah kepada mereka!
```

or

```suggestion
Mulailah dengan menggambar kotak-kotak di sekitar setiap komponen dan subkomponen dalam rancang bangun dan beri mereka nama. Jika Anda bekerja dengan seorang desainer, mereka mungkin telah menamai komponen-komponen ini di alat bantu desain mereka. Tanyakanlah kepada mereka!
```

or

```suggestion
Mulailah dengan menggambar kotak-kotak di sekitar setiap komponen dan subkomponen dalam rancang bangun dan berikan mereka nama. Jika Anda bekerja dengan seorang desainer, mereka mungkin telah menamai komponen-komponen ini di alat bantu desain mereka. Tanyakanlah kepada mereka!
```

**Review by mhaidarhanif — CHANGES_REQUESTED** [review]
Another suggestion is to make the reading flow better.

File: src/content/learn/thinking-in-react.md [hunk 6]
```diff
@@ -195,93 +195,93 @@ td {
...
-For each piece of state in your application:
+Untuk setiap bagian state dalam aplikasi Anda:
 
-1. Identify *every* component that renders something based on that state.
-2. Find their closest common parent component--a component above them all in the hierarchy.
-3. Decide where the state should live:
-    1. Often, you can put the state directly into their common parent.
-    2. You can also put the state into some component above their common parent.
-    3. If you can't find a component where it makes sense to own the state, create a new component solely for holding the state and add it somewhere in the hierarchy above the common parent component.
+1. Identifikasi *setiap* komponen yang merender sesuatu berdasarkan state tersebut.
```
Comment (r17x) [inline src/content/learn/thinking-in-react.md:257]: `merender` -> `me-render`

**Review by r17x — CHANGES_REQUESTED** [review]

**Review by mhaidarhanif — APPROVED** [review]
Looks ready, but maybe there are some things to check again. Wait for the others.

File: src/content/learn/thinking-in-react.md [hunk 7]
```diff
@@ -25,59 +25,59 @@ The JSON API returns some data that looks like this:
...
 
-To build a static version of your app that renders your data model, you'll want to build [components](/learn/your-first-component) that reuse other components and pass data using [props.](/learn/passing-props-to-a-component) Props are a way of passing data from parent to child. (If you're familiar with the concept of [state](/learn/state-a-components-memory), don't use state at all to build this static version. State is reserved only for interactivity, that is, data that changes over time. Since this is a static version of the app, you don't need it.)
+Untuk membuat versi statis dari aplikasi Anda yang merender model data Anda, Anda perlu membuat [komponen](/learn/your-first-component) yang menggunakan kembali komponen lain dan mengirimkan data menggunakan [props.](/learn/passing-props-to-a-component) Props adalah cara untuk mengoper data dari induk ke anak. (Jika Anda sudah terbiasa dengan konsep [state](/learn/state-a-components-memory), jangan gunakan state sama sekali untuk membangun versi statis ini. State hanya diperuntukkan bagi interaktivitas, yaitu data yang berubah seiring waktu. Karena ini adalah versi statis dari aplikasi, Anda tidak memerlukannya).
```
Comment (r17x) [inline src/content/learn/thinking-in-react.md:78]:
```suggestion
Untuk membuat versi statis dari aplikasi Anda yang me-*render* model data Anda, Anda perlu membuat [komponen](/learn/your-first-component) yang menggunakan kembali komponen lain dan mengirimkan data menggunakan [props.](/learn/passing-props-to-a-component) Props adalah cara untuk mengoper data dari induk ke anak. (Jika Anda sudah terbiasa dengan konsep [state](/learn/state-a-components-memory), jangan gunakan state sama sekali untuk membangun versi statis ini. State hanya diperuntukkan bagi interaktivitas, yaitu data yang berubah seiring waktu. Karena ini adalah versi statis dari aplikasi, Anda tidak memerlukannya).
```

File: src/content/learn/thinking-in-react.md [hunk 8]
```diff
@@ -195,93 +195,93 @@ td {
 
 </Sandpack>
 
-(If this code looks intimidating, go through the [Quick Start](/learn/) first!)
+(Jika kode ini terlihat menyeramkan, bacalah [Quick Start](/learn/) terlebih dahulu!)
 
-After building your components, you'll have a library of reusable components that render your data model. Because this is a static app, the components will only return JSX. The component at the top of the hierarchy (`FilterableProductTable`) will take your data model as a prop. This is called _one-way data flow_ because the data flows down from the top-level component to the ones at the bottom of the tree.
+Setelah membuat komponen, Anda akan memiliki pustaka komponen yang dapat digunakan kembali untuk merender model data Anda. Karena ini adalah aplikasi statis, komponen-komponennya hanya akan mengembalikan JSX. Komponen di bagian atas hirarki (`FilterableProductTable`) akan mengambil model data Anda sebagai props. Ini disebut _aliran data satu arah (one-way data flow)_ karena data mengalir turun dari komponen tingkat atas ke komponen di bagian bawah pohon.
```
Comment (r17x) [inline src/content/learn/thinking-in-react.md:200]:
```suggestion
Setelah membuat komponen, Anda akan memiliki pustaka komponen yang dapat digunakan kembali untuk me-*render* model data Anda. Karena ini adalah aplikasi statis, komponen-komponennya hanya akan mengembalikan JSX. Komponen di bagian atas hirarki (`FilterableProductTable`) akan mengambil model data Anda sebagai *props*. Ini disebut *aliran data satu arah (one-way data flow)* karena data mengalir turun dari komponen tingkat atas ke komponen di bagian bawah pohon.
```

File: src/content/learn/thinking-in-react.md [hunk 9]
```diff
@@ -25,59 +25,59 @@ The JSON API returns some data that looks like this:
...
         * `ProductRow`
 
-## Step 2: Build a static version in React {/*step-2-build-a-static-version-in-react*/}
+## Langkah 2: Buat versi statis di React {/*step-2-build-a-static-version-in-react*/}
 
-Now that you have your component hierarchy, it's time to implement your app. The most straightforward approach is to build a version that renders the UI from your data model without adding any interactivity... yet! It's often easier to build the static version first and add interactivity later. Building a static version requires a lot of typing and no thinking, but adding interactivity requires a lot of thinking and not a lot of typing.
+Setelah Anda memiliki hierarki komponen, sekarang saatnya mengimplementasikan aplikasi Anda. Pendekatan yang paling mudah adalah membuat versi yang merender UI dari model data Anda tanpa menambahkan interaktivitas apa pun... untuk sementara! Sering kali lebih mudah untuk membuat versi statis terlebih dahulu dan menambahkan interaktivitas kemudian. Membangun versi statis membutuhkan banyak pengetikan dan tidak perlu berpikir, tetapi menambahkan interaktivitas membutuhkan banyak pemikiran dan tidak perlu banyak pengetikan.
```
Comment (r17x) [inline src/content/learn/thinking-in-react.md:76]:
```suggestion
Setelah Anda memiliki hierarki komponen, sekarang saatnya mengimplementasikan aplikasi Anda. Pendekatan yang paling mudah adalah membuat versi yang me-*render* UI dari model data Anda tanpa menambahkan interaktivitas apa pun... untuk sementara! Sering kali lebih mudah untuk membuat versi statis terlebih dahulu dan menambahkan interaktivitas kemudian. Membangun versi statis membutuhkan banyak pengetikan dan tidak perlu berpikir, tetapi menambahkan interaktivitas membutuhkan banyak pemikiran dan tidak perlu banyak pengetikan.
```

File: src/content/learn/thinking-in-react.md (hunk 6 above)
Comment (r17x) [inline src/content/learn/thinking-in-react.md:257]:
```suggestion
1. Identifikasi *setiap* komponen yang me-*render* sesuatu berdasarkan *state* tersebut.
```

File: src/content/learn/thinking-in-react.md [hunk 10]
```diff
@@ -195,93 +195,93 @@ td {
...
 
-1. Identify *every* component that renders something based on that state.
-2. Find their closest common parent component--a component above them all in the hierarchy.
-3. Decide where the state should live:
-    1. Often, you can put the state directly into their common parent.
-    2. You can also put the state into some component above their common parent.
-    3. If you can't find a component where it makes sense to own the state, create a new component solely for holding the state and add it somewhere in the hierarchy above the common parent component.
+1. Identifikasi *setiap* komponen yang merender sesuatu berdasarkan state tersebut.
+2. Temukan komponen induk yang paling dekat--komponen yang berada di atas semua komponen dalam hirarki.
+3. Tentukan di mana state tersebut harus berada:
```
Comment (r17x) [inline src/content/learn/thinking-in-react.md:259]:
```suggestion
3. Tentukan di mana *state* tersebut harus berada:
```

File: src/content/learn/thinking-in-react.md [hunk 11]
```diff
@@ -195,93 +195,93 @@ td {
...
+    1. Sering kali, Anda dapat meletakkan state secara langsung ke dalam induknya.
```
Comment (r17x) [inline src/content/learn/thinking-in-react.md:260]:
```suggestion
    1. Sering kali, Anda dapat meletakkan *state* secara langsung ke dalam induknya.
```

File: src/content/learn/thinking-in-react.md [hunk 12]
```diff
@@ -195,93 +195,93 @@ td {
...
+    2. Anda juga dapat menempatkan state ke dalam beberapa komponen di atas induknya.
```
Comment (r17x) [inline src/content/learn/thinking-in-react.md:261]:
```suggestion
    2. Anda juga dapat menempatkan *state* ke dalam beberapa komponen di atas induknya.
```

File: src/content/learn/thinking-in-react.md [hunk 13]
```diff
@@ -195,93 +195,93 @@ td {
...
+    3. Jika Anda tidak dapat menemukan komponen yang masuk akal untuk memiliki state, buatlah komponen baru hanya untuk menyimpan state dan tambahkan di suatu tempat di dalam hirarki di atas komponen induk umum.
```
Comment (r17x) [inline src/content/learn/thinking-in-react.md:262]:
```suggestion
    3. Jika Anda tidak dapat menemukan komponen yang masuk akal untuk memiliki *state*, buatlah komponen baru hanya untuk menyimpan *state* dan tambahkan di suatu tempat di dalam hirarki di atas komponen induk umum.
```

File: src/content/learn/thinking-in-react.md [hunk 14]
```diff
@@ -437,15 +437,15 @@ td {
...
+Perhatikan bahwa pengeditan formulir belum berhasil. Ada galat di konsol di *sandbox* di atas yang menjelaskan alasannya:
 
 <ConsoleBlock level="error">
 
 You provided a \`value\` prop to a form field without an \`onChange\` handler. This will render a read-only field.
 
 </ConsoleBlock>
 
-In the sandbox above, `ProductTable` and `SearchBar` read the `filterText` and `inStockOnly` props to render the table, the input, and the checkbox. For example, here is how `SearchBar` populates the input value:
+Pada *sandbox* di atas, `ProductTable` dan `SearchBar` membaca props `filterText` dan `inStockOnly` untuk merender tabel, input, dan kotak centang. Sebagai contoh, berikut ini cara `SearchBar` mengisi nilai input:
```
Comment (r17x) [inline src/content/learn/thinking-in-react.md:448]:
```suggestion
Pada *sandbox* di atas, `ProductTable` dan `SearchBar` membaca props `filterText` dan `inStockOnly` untuk me-*render* tabel, input, dan kotak centang. Sebagai contoh, berikut ini cara `SearchBar` mengisi nilai input:
```

**Review by r17x — CHANGES_REQUESTED** [review]
* word for "state" -> `*state*`
* word for "merender" -> `me-*render*`

**Review by r17x — APPROVED** [review]

**Pull request merged**